### PR TITLE
Get the width from integer and float type

### PR DESCRIPTION
**Pull request by Qining**

### CODE REVIEW

File: source/opt/types.h [hunk 1]
```diff
@@ -123,14 +123,15 @@ class Type {
...
-  Integer(uint32_t width, bool is_signed) : width_(width), signed_(is_signed) {}
+  Integer(uint32_t w, bool is_signed) : width_(w), signed_(is_signed) {}
   Integer(const Integer&) = default;
 
   bool IsSame(Type* that) const override;
   std::string str() const override;
 
   Integer* AsInteger() override { return this; }
   const Integer* AsInteger() const override { return this; }
+  uint32_t width() const {return width_;}
```
Comment (antiagainst) [inline source/opt/types.h]: clang-format

File: test/opt/test_types.cpp [hunk 2]
```diff
@@ -218,4 +218,20 @@ TEST(Types, AllTypes) {
   }
 }
 
+#define TestWidthOfType(ty, ...)                                      \
+  TEST(Types, TypeWidthOf##ty) {                                      \
+    std::vector<uint32_t> widths = {1, 2, 4, 8, 16, 32, 48, 64, 128}; \
+    std::vector<std::unique_ptr<ty>> types;                           \
+    for (uint32_t w : widths) {                                       \
+      types.emplace_back(new ty(w, ##__VA_ARGS__));                   \
```
Comment (antiagainst) [inline test/opt/test_types.cpp]: Ha, I'm afraid this `##__VA_ARGS__` trick suffers on some old VS compilers? (Although we only test >= VS2013 in buildbot, it's still better to be able to compile on older compilers.)

I'm not opposed to duplicate the code for Integer & Float anyway, it's just two cases.

Comment (antiagainst) [inline test/opt/test_types.cpp]: http://stackoverflow.com/questions/5588855/standard-alternative-to-gccs-va-args-trick

**Comment by Qining** [comment]
All comments fixed in 6c5bdc9

**Comment by antiagainst** [comment]
LGTM

**Pull request merged**